mautic/mautic
# Block access to contact information updated

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | 🟢 
| New feature/enhancement? (use the a.x branch)      | 🔴
| Deprecations?                          | 🔴
| BC breaks? (use the c.x branch)        | 🔴
| Automated tests included?              | 🟢 
| Related user documentation PR URL      | mautic/user-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation-new#... 
| Issue(s) addressed                     | Fixes #... 



## Description

This PR fixes the handling of unexpected exceptions and the access to contact information without the necessary permissions.





| Before                                 | After
| -------------------------------------- | ---
| ![before](https://github.com/user-attachments/assets/198c864b-73d4-4603-add5-b7b276d40227) |  ![after](https://github.com/user-attachments/assets/93b663e5-a2fb-44f3-b93d-a31165078fb9)


---
### 📋 Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Remove all contact permissions (`email:emails:viewown`, `email:emails:viewother`)
3. Navigate to a segment details view, click on contacts
4. The contacts included in the segment are visible

The contact grid is permission is updated for Campaign and Email details page, out of which Email pages has the issue as mentioned in the description so updated testing steps are,
1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create a non-admin user with the following permission
   1. Email: View Own, View Others, Create
   2. Contacts:  View Own, View Others, Create
3. Create contacts with different owners and create segments for these contact
4. Create a broadcast or segment-type email and select the above-created segment.
5. Run the broadcast
6. Navigate to the email details page and click on the Contact tab
7. The contact grid will start showing up
